smclient/smclient.go

```diff
@@ -187,24 +182,17 @@ func New(config *config.Config, nodeInfoProvider NodeInfoProvider, certificatePr
 		cmClient.alertChannel = alertsProvider.GetAlertsChannel()
 	}
 
+	if monitoringProvider != nil {
```

Comment:
It's not clear how cmClient.monitoringChannel is utilized by smClient

Reply:
There is the private member `monitoringChannel`. It is handled in the separate thread. Before, we put items into this channel in `SendNodeMonitoring`. Now, we directly handle node monitoring channel from `MonitoringProvider`.